### PR TITLE
Async map icons DT-1353

### PR DESCRIPTION
**Pull request by samuliheljo**
The PR is done - it:

- [ ] follows the style and naming rules (passes `npm run lint`)
- [ ] doesn't break anything (passes `npm run test-local` and `npm run test-browserstack`)
- [ ] design is as expected. NOTE! visuals are compared using HSL theme (`CONFIG=hsl npm run dev`).
-- If no design changes: `BS_USERNAME=user BS_ACCESS_KEY=key npm run test-visual` passes
-- If design changes: `BS_USERNAME=user BS_ACCESS_KEY=key npm run test-visual-update` to generate new images
- [ ] any changed files are transformed to ES6
- [ ] all new strings visible to users are

  * [ ] added to translations file
  * [ ] are translated to English, Finnish and Swedish (if not, add translations-needed label to PR)
  
- [ ] all changed components

  * [ ] have examples
  * [ ] are included in the style guide
  * [ ] are included in the visual tests (added to gemini tests)

If this PR fixes a bug, it includes a new test that catches the bug to prevent regressions.

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -16,6 +16,7 @@ module.exports = {
       'env': 'relay',
       'schemaJson': require('./build/schema.json'),
     }],
+    'generator-star-spacing': 0,
```
Comment (hannesj) [inline .eslintrc.js]: Why is this needed?

File: app/component/map/tile-layer/CityBikes.js [hunk 2]
```diff
@@ -3,7 +3,14 @@ import Protobuf from 'pbf';
 import Relay from 'react-relay';
 import glfun from 'mapbox-gl-function';
 import config from '../../../config';
-import { drawRoundIcon, getImageFromSprite } from '../../../util/mapIconUtils';
+import {
```
Comment (hannesj) [inline app/component/map/tile-layer/CityBikes.js]: Could there be a function called drawAvailabilityBadge, where you give the availability as a parameter?

File: .eslintrc.js (hunk 1 above)
Comment (samuliheljo) [inline .eslintrc.js]: I'm not sure if there is some better way to handle this: https://github.com/babel/babel-eslint/issues/316

Comment (hannesj) [inline .eslintrc.js]: Oh, I see. Could you add that as a comment?

File: app/util/mapIconUtils.js [hunk 3]
```diff
@@ -78,107 +128,111 @@ export function drawRoundIcon(tile, geom, type, large, platformNumber) {
...
-    tile.ctx.fillStyle = gradient;
-    tile.ctx.fillRect(
-      (geom.x / tile.ratio) - (haloRadius * tile.scaleratio),
-      (geom.y / tile.ratio) - (haloRadius * tile.scaleratio),
-      haloRadius * tile.scaleratio * 2,
-      haloRadius * tile.scaleratio * 2
+  if (stopRadius <= 0) return;
+
+  const iconSize = (stopRadius - 2) * tile.scaleratio;
+  const image = await getImageFromSpriteAsync('icon-icon_station', iconSize, iconSize, 'white');
```
Comment (hannesj) [inline app/util/mapIconUtils.js]: Should this be `getImageFromSpriteCache`?
